File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {boolean}
     */
    hasCycle(head) {
            let current=head
        while(current)
        {
            if(current.visited) return true
            current.visited=true
            current=current.next
        }

        return false
    }
}

File: Data Structures & Algorithms/reverse-linked-list-ii/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 
func reverseBetween(head *ListNode, left int, right int) *ListNode {
    dummy:=&ListNode{Next:head}
	leftPrev:=dummy

	for i:=1;i<left;i++ {
		leftPrev=leftPrev.Next
	}

	current:=leftPrev.Next

	var prev *ListNode

	for i:=1;i<=right-left+1;i++ {
		next:=current.Next
		current.Next=prev
		prev=current
		current=next
	}
	leftPrev.Next.Next=current
	leftPrev.Next=prev


	return dummy.Next
}
